packages/core/src/verification/signature-verifier.ts

```diff
@@ -0,0 +1,543 @@
+
+          if (!response.ok) {
+            throw new Error(
+              'Failed to fetch data item info from trusted gateway',
+              {
+                cause: {
+                  txId,
+                  gateway: gateway.toString(),
+                },
+              },
+            );
+          }
+
+          const dataItemOffset = response.headers.get(
+            arioGatewayHeaders.dataItemOffset,
+          );
+          const dataItemDataOffset = response.headers.get(
+            arioGatewayHeaders.dataItemDataOffset,
+          );
+          const rootTransactionId = response.headers.get(
+            arioGatewayHeaders.rootTransactionId,
+          );
+          const dataItemSize = response.headers.get(
+            arioGatewayHeaders.dataItemSize,
+          );
+
+          // if any are undefined, throw an error
+          if (
+            dataItemOffset &&
+            dataItemDataOffset &&
+            rootTransactionId &&
+            dataItemSize
+          ) {
+            throttle.clearQueue();
+            return {
+              gateway,
+              dataItemOffset: parseInt(dataItemOffset, 10),
+              dataItemDataOffset: parseInt(dataItemDataOffset, 10),
+              rootTransactionId,
+              dataItemSize: parseInt(dataItemSize, 10),
+            };
+          }
+
+          // if we get here, we didn't get the information we need from this gateway
+          throw new Error('Missing data item attributes from trusted gateway', {
+            cause: {
+              txId,
+              gateway: gateway.toString(),
+              dataItemOffset,
+              dataItemDataOffset,
+              rootTransactionId,
+              dataItemSize,
+            },
+          });
+        });
+      },
+    );
+
+    const {
+      gateway,
+      dataItemOffset,
+      dataItemDataOffset,
+      rootTransactionId,
+      dataItemSize,
+    } = await Promise.any(dataItemAttributesPromises);
+    this.logger.debug('Successfully fetched data item attributes for txId', {
+      txId,
+      dataItemOffset,
+      dataItemDataOffset,
+      rootTransactionId,
+      dataItemSize,
+      gateway: gateway.toString(),
+    });
+    return {
+      dataItemOffset,
+      dataItemDataOffset,
+      rootTransactionId,
+      dataItemSize,
+    };
+  }
+
+  /**
+   * Fetches just the bytes needed for signature verification from the root transaction
+   *
+   * @param txId - The transaction ID to get signature data for
+   * @returns The bytes containing the data item up to the data section
+   */
+  async getSignatureData({
+    txId,
+  }: {
+    txId: string;
+  }): Promise<{
+    signatureType: number;
+    signature: Uint8Array;
+    owner: Uint8Array;
+    target: Uint8Array;
+    anchor: Uint8Array;
+    tags: Uint8Array;
+  }> {
+    const { rootTransactionId, dataItemOffset, dataItemDataOffset } =
+      await this.getDataItemAttributes(txId);
+
+    // Calculate the byte range for the request
```

Comment:
Nit: Took a minute to realize this section was fetching the header bytes of the data item via offsets. Might want to explain that a bit more or make it more obvious using a named function.